apps/studio/src/server/modules/site/site.router.ts

```diff
@@ -157,16 +170,46 @@ export const siteRouter = router({
       })
 
       await db.transaction().execute(async (tx) => {
-        await tx
+        const user = await tx
+          .selectFrom("User")
+          .where("id", "=", ctx.user.id)
+          .selectAll()
+          .executeTakeFirstOrThrow()
+
+        // Update site-level configuration
+        const oldSite = await tx
+          .selectFrom("Site")
+          .where("id", "=", siteId)
+          .selectAll()
+          .executeTakeFirstOrThrow()
+
+        const newSite = await tx
           .updateTable("Site")
           .set({
             config: jsonb(safeJsonParse(config) as IsomerSiteConfigProps),
             theme: jsonb(safeJsonParse(theme) as IsomerSiteThemeProps),
           })
           .where("id", "=", siteId)
-          .execute()
+          .returningAll()
+          .executeTakeFirstOrThrow()
 
-        await tx
+        await logConfigEvent(tx, {
+          eventType: AuditLogEvent.SiteConfigUpdate,
+          delta: {
+            before: oldSite,
+            after: newSite,
+          },
+          by: user,
+        })
+
+        // Update Navbar contents
+        const oldNavbar = await tx
+          .selectFrom("Navbar")
+          .where("siteId", "=", siteId)
+          .selectAll()
+          .executeTakeFirstOrThrow()
```

Comment:
here

Reply:
Updated in 489502187e9c9f571dc75a6b344c5ea98ce31990